basic_module/staking/src/state.rs

```diff
@@ -0,0 +1,540 @@
+// Copyright 2020 Kodebox, Inc.
+// This file is part of CodeChain.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+use crate::error::{Insufficient, Mismatch};
+use crate::runtime_error::Error;
+use crate::types::*;
+use crate::{deserialize, serialize, substorage};
+use serde::{de::DeserializeOwned, ser::Serialize};
+use std::cmp::{max, Ordering};
+use std::collections::{
+    btree_map::{self, Entry},
+    BTreeMap, BTreeSet, HashMap, HashSet,
+};
+use std::ops::Deref;
+
+type KEY = dyn AsRef<[u8]>;
+
+const STAKE_ACCOUNT_PREFIX: [u8; 1] = [0x1];
+const DELEGATION_PREFIX: [u8; 1] = [0x2];
+lazy_static! {
+    static ref METADATA_KEY: Vec<u8> = String::from("Metadata").into_bytes();
```

Comment:
Are these should be vectors?
If not, you don't need to use `lazy_static`.
```rust
static METADATA_KEY: &[u8; 8] = b"Metadata";
```

Reply:
Ah, thank you. I'll apply your suggestion. They don't have to be vectors.